app/src/main/java/com/devshish/internship/adapter/ItemSongAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.devshish.internship.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.devshish.internship.databinding.ItemSongBinding
+import com.devshish.internship.model.Song
+
+class ItemSongAdapter : RecyclerView.Adapter<ItemSongAdapter.SongViewHolder>() {
```

Comment:
Also - read description:
You should implement ListAdapter here , not just simple RecyclerView.Adapter